### ISSUE TITLE
check_source: filename check does not work on packages targeting images repository

### ISSUE THREAD

**Issue by jberry-suse** [openSUSE/openSUSE-release-tools#2244]
After #2231, goes from a decline, which on it's own likely needs some followup to package or tooling, to a crash for [request 727561 for livecd-openSUSE](https://build.opensuse.org/request/show/727561).

```
$ ./check_source.py --dry --debug id 727561
[I] checking 727561
[I] openSUSE:Factory/livecd-openSUSE@59 -> openSUSE:Leap:15.2/livecd-openSUSE
[W] directory /home/jberry/co/727561 already exists
Traceback (most recent call last):
  File "/home/jberry/openSUSE-release-tools/ReviewBot.py", line 187, in check_requests
    good = self.check_one_request(req)
  File "/home/jberry/openSUSE-release-tools/ReviewBot.py", line 408, in check_one_request
    ret = func(req, a)
  File "/home/jberry/openSUSE-release-tools/ReviewBot.py", line 500, in check_action_submit
    return self.check_source_submission(a.src_project, a.src_package, a.src_rev, a.tgt_project, a.tgt_package)
  File "./check_source.py", line 154, in check_source_submission
    if not new_info['filename'].endswith('.kiwi') and new_info['name'] != target_package:
KeyError: 'filename'
[I] 727561 ignored
```

as `new_info` is `{'name': None, 'version': None}`.

**Comment by jberry-suse** [comment]
An aside, can see the series of requests from Factory that were being declined before the change to crash:

![image](https://user-images.githubusercontent.com/22943929/65645878-c1388d80-dfbe-11e9-8b7d-73c81f665ba2.png)

**Comment by lnussel** [comment]
Not sure why obs behaves that way. This one works:
osc api '/source/openSUSE:Factory:Live/livecd-openSUSE?view=info&parse=1'

This one doesn't
osc api '/source/openSUSE:Factory/livecd-openSUSE?view=info&parse=1'

Even though the former is just a link. @mls?

**Comment by mlschroe** [comment]
The error message pretty much gives it away:
```
<sourceinfo package="livecd-openSUSE" rev="72" vrev="72" srcmd5="2349d5ad0980e9576f4a0bc6d4bdf319" verifymd5="e0012531130c4f47b02ad8ba18e2c563">
  <error>no file found for build type 'spec'</error>
</sourceinfo>
```

You're not providing a repository, so the build service just uses the project config of the specified project instead of doing the normal repository path expansion. And `%_repository` is also not set.

You need to specify a repository to make it work:
`osc api '/source/openSUSE:Factory/livecd-openSUSE?view=info&parse=1&repository=images'`

**Comment by lnussel** [comment]
that's all Greek to me. how is factory-auto supposed to know which repo to use?
So the effect of that is that for some reason a submission from
Virtualization:Appliances:Images:openSUSE-Tumbleweed/livecd-openSUSE as source works and a submission from Factory not.

**Comment by jberry-suse** [comment]
So essentially the trick is that we need the following prjconf condition to set the type:

```
%if "%_repository" == "images"
Type: kiwi
Repotype: none
Patterntype: none
Prefer: openSUSE-release-ftp
%endif
```

The problem is looking at the package alone there is no clean way to know to do that. I'd suggest the best route (to avoid hard-coding) would be try call as is (or use the `main-repo` arg in config which is set to `standard` for Factory/Leap/SLE) and if `filename` is not set then try all repos (or all other) and assuming one validates the rule accept.

**Comment by jberry-suse** [comment]
From repo-checker development I would expect there to be relevant helpers for most if not all the repository stuff in `osclib.core`.

**Comment by lnussel** [comment]
out of scope for me. #2250 restores previous behavior
